includes/woopay/class-woopay-adapted-extensions.php

```diff
@@ -171,9 +171,38 @@ public function get_extension_data() {
 			];
 		}
 
+		if ( $this->is_affiliate_for_woocommerce_enabled() ) {
+			$extension_data[ 'affiliate-for-woocommerce' ] = [
+				'affiliate-user' => afwc_get_referrer_id(),
+			];
+		}
+
 		return $extension_data;
 	}
 
+	/**
+	 * Update order extension data after finishing
+	 * an order on WooPay, this usually is needed
+	 * for extensions which uses cookies when an
+	 * order is finished.
+	 *
+	 * @param int $order_id The successful WooPay order.
+	 */
+	public function update_order_extension_data( $order_id ) {
+		if ( ! empty( $_GET['affiliate'] ) && // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
It's unlikely, but this could happen on any store API endpoint with an affiliate id in the URL.  Should we limit it to just the checkout endpoint?

Reply:
This code is only executed when the request comes from WooPay, but I've added a new check on the WooPay side to only add the parameter on the checkout route request.